src/main/native/cpp/view/Analyzer.cpp

```diff
@@ -587,3 +536,88 @@ void Analyzer::ResetManagerState() {
   m_manager.reset();
   *m_location = "";
 }
+
+void Analyzer::LoadFeedForwardGains(bool combined) {
+  float beginX = ImGui::GetCursorPosX();
+  float beginY = ImGui::GetCursorPosY();
+  const char* gainNames[] = {"Ks", "Kv", "Ka"};
+  for (size_t i = 0; i < 3; i++) {
+    SetPosition(beginX, beginY, 0, i);
+    ShowGain(gainNames[i], &m_ff[i]);
+  }
+
+  SetPosition(beginX, beginY, 0, 3);
+
+  if (m_type == analysis::kElevator) {
+    ShowGain("Kg", &m_ff[3]);
+  } else if (m_type == analysis::kArm) {
+    ShowGain("Kcos", &m_ff[3]);
+  } else if (m_trackWidth) {
+    ShowGain("Track Width", &*m_trackWidth);
+  }
+
+  SetPosition(beginX, beginY, 0, 4);
+  ShowGain("r-squared", &m_rSquared);
+
+  SetPosition(beginX, beginY, 0, 5);
+  ShowGain("Sim RMSE", m_plot.GetRMSE());
+
+  if (!combined) {
+    CreateTooltip(
+        "The Root Mean Squared Error (RMSE) of the simulation "
+        "predictions compared to the recorded data. It is essentially the "
+        "error of the simulated model in the recorded units.");
+  }
+
+  double endY = ImGui::GetCursorPosY();
+  SetPosition(beginX, beginY, 1, 0);
+
+  if (!combined) {
+    if (ImGui::Button("Combined Diagnostics")) {
+      ImGui::OpenPopup("Combined Diagnostics");
+    }
+    SetPosition(beginX, beginY, 1, 1);
+  }
+
+  ImGui::SetNextItemWidth(ImGui::GetFontSize() * 4);
+  int window = m_settings.windowSize;
+  if (ImGui::InputInt(
+          "Window Size", &window, 0, 0,
+          combined ? ImGuiInputTextFlags_ReadOnly : ImGuiInputTextFlags_None)) {
+    m_settings.windowSize = std::clamp(window, 2, 15);
+    RefreshInformation();
+  }
+
+  SetPosition(beginX, beginY, 1, combined ? 1 : 2);
+  ImGui::SetNextItemWidth(ImGui::GetFontSize() * 4);
+  double threshold = m_settings.motionThreshold;
+  if (ImGui::InputDouble(
+          "Velocity Threshold", &threshold, 0.0, 0.0, "%.3f",
+          combined ? ImGuiInputTextFlags_ReadOnly : ImGuiInputTextFlags_None)) {
+    m_settings.motionThreshold = std::max(0.0, threshold);
+    RefreshInformation();
+  }
+
+  SetPosition(beginX, beginY, 1, combined ? 2 : 3);
+  ImGui::SetNextItemWidth(ImGui::GetFontSize() * 4);
+  if (!combined) {
+    if (ImGui::SliderFloat("Test Duration", &m_stepTestDuration,
+                           m_manager->GetMinDuration(),
+                           m_manager->GetMaxDuration(), "%.2f")) {
+      m_settings.stepTestDuration = units::second_t{m_stepTestDuration};
+      RefreshInformation();
+    }
+
+    ImGui::SetCursorPosY(endY);
+  } else {
+    std::ostringstream ss;
+    ss << std::round(m_stepTestDuration * 100) / 100 << " of "
```

Comment:
Use `wpi::format` from `<wpi/Format.h>` instead.

Reply:
Is that similar to the c++20 std::format?

Reply:
Not really. fmtlib is more like C++20 std::format.

Reply:
You can do `wpi::format(v, "%.2f")`

Reply:
Is it properly implemented?